core/src/main/java/com/oath/oak/InternalOakMap.java

```diff
@@ -1101,7 +1107,8 @@ public int getIndex() {
             this.hiInclusive = hiInclusive;
             this.isDescending = isDescending;
 
-            initState(isDescending, lo, loInclusive, hi, hiInclusive);
+            ThreadContext ctx = getThreadLocalContext();
```

Comment:
How about iterating thread doing something else in between the iterations and invalidating/changing the thread context ? Might it break something? We should add such test.

Reply:
This context is used only in the constructor of the iterator. It is similar to any other operation.